Validate with an invalid table schema fails with AttributeError

# Overview

```

________________________________________________ test_validate_invalid_table_schema ________________________________________________

log = <function log.<locals>.fixture at 0x7f21677a8d08>

    def test_validate_invalid_table_schema(log):
        source = [
            ['name', 'age'],
            ['Alex', '33'],
        ]
        schema = {'fields': [
            {'name': 'name'},
            {'name': 'age', 'type': 'bad'},
        ]}
>       report = validate(source, schema=schema)

tests/test_validate.py:94:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
goodtables/validate.py:85: in validate
    report = inspector.inspect(source, **options)
goodtables/inspector.py:68: in inspect
    warnings, tables = preset_func(source, **options)
goodtables/presets/table.py:29: in table
    schema = Schema(schema)
.python/lib/python3.7/site-packages/tableschema/schema.py:41: in __init__
    self.__build()
.python/lib/python3.7/site-packages/tableschema/schema.py:262: in __build
    field = Field(field, missing_values=missing_values)
.python/lib/python3.7/site-packages/tableschema/field.py:32: in __init__
    self.__cast_function = self.__get_cast_function()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <tableschema.field.Field object at 0x7f2166662748>

    def __get_cast_function(self):
        options = {}
        # Get cast options
        for key in ['decimalChar', 'groupChar', 'bareNumber', 'trueValues', 'falseValues']:
            value = self.descriptor.get(key)
            if value is not None:
                options[key] = value
>       cast = getattr(types, 'cast_%s' % self.type)
E       AttributeError: module 'tableschema.types' has no attribute 'cast_bad'

.python/lib/python3.7/site-packages/tableschema/field.py:122: AttributeError
```

--- a/setup.py
+++ b/setup.py
@@ -28,7 +28,7 @@ INSTALL_REQUIRES = [
     'simpleeval>=0.9',
     'statistics>=1.0',
     'tabulator>=1.29',
-    'tableschema>=1.16',
+    'tableschema>=1.16.1',
     'datapackage>=1.10',
 ]
 INSTALL_FORMAT_ODS_REQUIRES = [

--- a/tests/test_validate.py
+++ b/tests/test_validate.py
@@ -431,3 +431,16 @@ def test_validate_fails_with_wrong_encoding_issue_274(log):
     # For now, by default encoding is detected incorectly by chardet
     report = validate('data/encoding-274.csv', encoding='utf-8')
     assert report['valid']
+
+
+def test_validate_invalid_table_schema_issue_304(log):
+    source = [
+        ['name', 'age'],
+        ['Alex', '33'],
+    ]
+    schema = {'fields': [
+        {'name': 'name'},
+        {'name': 'age', 'type': 'bad'},
+    ]}
+    report = validate(source, schema=schema)
+    assert not report['valid']

